gym_xiangqi/test/xiangqi_env_test.py

```diff
@@ -11,15 +17,61 @@ def test_env_initialization(self):
         self.assertEqual(env.agent_color, RED)
         self.assertEqual(env.enemy_color, BLACK)
 
-        env = XiangQiEnv(agent_color=1)
+        env = XiangQiEnv(agent_color=BLACK)
         self.assertEqual(env.agent_color, BLACK)
         self.assertEqual(env.enemy_color, RED)
 
         self.assertIsNotNone(env.game)
         self.assertIsNotNone(env.state)
         self.assertIsNotNone(env.observation_space)
         self.assertIsNotNone(env.action_space)
-        self.assertIsNotNone(env.possible_actions)
+        self.assertIsNotNone(env.agent_actions)
+        self.assertIsNotNone(env.enemy_actions)
+
+    def test_env_step(self):
+        env = XiangQiEnv()
+
+        # verify action input
+        with self.assertRaises(AssertionError):
+            env.step(-1)
+
+        # verify agent with illegal move
+        obs = env.state
+        new_obs, reward, done, info = env.step(0)
+
+        for i in range(BOARD_ROWS):
+            for j in range(BOARD_COLS):
+                self.assertEqual(obs[i][j], obs[i][j])
```

Comment:
Does it work between two lists? I was not sure because they are not values. Still they do not even point to the same data structure in memory because `step()` returns observation as a copy not the address to original np array.

Reply:
Oh I meant a typo cause now you are doing `self.assertEqual(obs[i][j], obs[i][j])`

Reply:
ohh oops lol